### ISSUE TITLE
Add support in CalloutSourceRx for CSS comments?

### ISSUE THREAD

**Issue by sandersk** [asciidoctor/asciidoctor#1467]
The CalloutSourceRx regex:

https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor.rb#L748

Used for callout substitutions in code listings:

https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/substitutors.rb#L1118

Doesn't support CSS-style comments, e.g.:

````
[source, css]
----
p { color: green; } /*<1>*/
----
````

And while AsciiDoctor also provides a `line-comment` attribute to permit users to supply a custom comment delimiter, this delimiter is prepended to the beginning of the comment, but not the end, and is not compatible with comment syntaxes that require a start and ending delimiter (`/* like this in CSS */`).

Would it be in scope to either modify `CalloutSourceRx` to support CSS comments, or to add more functionality to the `line-comment` handling to support comment types that require start-end delimiters? The latter might make more sense, as it would potentially be flexible enough to accommodate other comment syntaxes besides just CSS.

**Comment by mojavelinux** [comment]
@sandersk It's good to hear from you!

Wow, I can't believe we forgot CSS. Like XML and HTML, it lacks line comments, so we would have to do the same trick we do for those languages. I don't think expanding the behavior of `line-comment` is the right approach because what we need to specify isn't a line comment.

I'm actually not happy with the current regexp approach because it's already fuzzy as it is (matches any comments in any language). This case only further demonstrates that the approach doesn't scale. I'd like to take the opportunity to make it right by honoring the rules of the language or language family. For languages that we don't recognize, we could have a `callout-guard` attribute that has a start and end value (e.g., `callout-guard="/*,*/`"). If only one value is present, it acts like the current `line-comment`.

Long story short, we've been trying to get away with a single regexp to handle this whole feature and I think it is time to switch to a more formal and correct approach.
